fe-local-treasures/screens/SignUpScreen.jsx

```diff
@@ -1,155 +1,155 @@
 import { StatusBar } from "expo-status-bar";
-import { useState } from "react";
-import { Text, View, ScrollView } from "react-native";
-import { Button, Input } from "react-native-elements";
-import Icon from "react-native-vector-icons/FontAwesome";
+import { useState, useEffect } from "react";
+import { View, ScrollView } from "react-native";
 import * as ImagePicker from "expo-image-picker";
 import { styles } from "../styles/signupScreens";
 
 import { getAuth, createUserWithEmailAndPassword } from "firebase/auth";
 import "firebase/auth";
-import ImageViewer from "./ImageViewer";
-import { textStyles } from "../styles/textStyles";
-import { buttons } from "../styles/buttons";
-import { globalStyles } from "../styles/globalStyles";
+import { getFirestore } from "firebase/firestore";
+import {
+  getStorage,
+  ref,
+  uploadBytesResumable,
+  getDownloadURL,
+} from "firebase/storage";
+import SignUpForm from "./components/SignUpForm";
 
 const auth = getAuth();
+const db = getFirestore();
+const storage = getStorage();
 
 export default function SignUpScreen({ navigation }) {
   const [selectedImage, setSelectedImage] = useState(null);
+  const [uploadProgress, setUploadProgress] = useState("");
+  const [progressBar, setProgressBar] = useState(0);
   const [userInfo, setUserInfo] = useState({
     username: "",
     firstname: "",
     lastname: "",
     email: "",
     password: "",
     error: "",
+    imgUrl: "",
   });
 
-  const pickImageAsync = async () => {
+  const pickImage = async () => {
     let result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.All,
       allowsEditing: true,
       quality: 1,
     });
-
     if (!result.canceled) {
       setSelectedImage(result.assets[0].uri);
     } else {
       alert("You did not select any image.");
     }
   };
 
-  function signUp() {
+  useEffect(() => {
+    const uploadImage = async () => {
+      const blobImage = await new Promise((resolve, reject) => {
+        const xhr = new XMLHttpRequest();
+        xhr.onload = function () {
+          resolve(xhr.response);
+        };
+        xhr.onerror = function () {
+          reject(new TypeError("Network request failed"));
+        };
+        xhr.responseType = "blob";
+        xhr.open("GET", selectedImage, true);
+        xhr.send(null);
+      });
+
+      /** @type {any} */
```

Comment:
remove this I think?

Reply:
This is part of the initial setup for the image upload